fvm/env.go

```diff
@@ -244,22 +245,48 @@ func (e *hostEnv) CacheProgram(location ast.Location, program *ast.Program) erro
 	return e.ctx.ASTCache.SetProgram(location, program)
 }
 
-func (e *hostEnv) Log(message string) {
-	e.logs = append(e.logs, message)
+func (e *hostEnv) Log(message string) error {
+	if e.ctx.CadenceLoggingEnabled {
+		e.logs = append(e.logs, message)
+	}
+	return nil
 }
 
-func (e *hostEnv) EmitEvent(event cadence.Event) {
-	e.events = append(e.events, event)
-}
+func (e *hostEnv) EmitEvent(event cadence.Event) error {
 
-func (e *hostEnv) GenerateUUID() uint64 {
-	uuid, err := e.uuidGenerator.GenerateUUID()
+	payload, err := jsoncdc.Encode(event)
```

Comment:
Should we really measure size by encoding to the very verbose JSON format?

Reply:
Unfortunately, that's how we store them right now and deliver them to users, so I kept the JSON encoding so users can have some measures of how big the events are. We increase the limit in the future when we store things more binary.

Reply:
Oh, I thought we store them as Cadence interpreter values, like in the account storage. It would be great if we could switch, I don't even think the JSON encoding is deterministic?

Reply:
> I don't even think the JSON encoding is deterministic
thats a good point, we def need to change for event verification